mongorestore/restore.go

```diff
@@ -17,17 +17,57 @@ import (
 	"github.com/mongodb/mongo-tools-common/log"
 	"github.com/mongodb/mongo-tools-common/progress"
 	"github.com/mongodb/mongo-tools-common/util"
+
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
 )
 
 const insertBufferFactor = 16
 
+// Result encapsulates the outcome of a particular restore attempt.
+type Result struct {
```

Comment:
mongoimport uses atomic variables to track success/failure counts.  Why isn't that an issue for restore?  Should we/could we have a similar approach (either way) for both programs?